test/ncengine/physics/Vehicle_unit_tests.cpp

```diff
@@ -0,0 +1,183 @@
+#include "jolt/JoltApiFixture.inl"
+#include "ncengine/physics/Vehicle.h"
+
+#include "physics/jolt/ConstraintFactory.h"
+
+#include "Jolt/Jolt.h"
+#include "Jolt/Physics/Vehicle/VehicleConstraint.h"
+#include "Jolt/Physics/Vehicle/WheeledVehicleController.h"
+
+class VehicleTest : public JoltApiFixture
+{
+    protected:
+        VehicleTest()
+            : constraintFactory{joltApi.physicsSystem}
+        {
+        }
+
+        ~VehicleTest()
+        {
+            constraints.clear();
+            for (auto body : bodies)
+            {
+                DestroyBody(body);
+            }
+        }
+
+    public:
+        nc::physics::ConstraintFactory constraintFactory;
+        std::vector<JPH::Body*> bodies;
+        std::vector<JPH::Ref<JPH::VehicleConstraint>> constraints;
+
+        auto CreateVehicle(nc::VehicleInfo&& info) -> nc::Vehicle
```

Comment:
Why && here? I'm sure you did it for a reason but I can't puzzle it out

Reply:
Eh, it doesn't have to be here for the test, but I was probably just trying to be consistent with `Vehicle`'s constructor. I made that one take by `VehicleInfo&&` because its moderately expensive to copy. Recommendation at [F.18](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-consume) of core guidelines. We sort of treat the 'exception' to that rule as the default because it makes for a more friendly interface.